src/lib/Validations.js

```diff
@@ -1,13 +1,14 @@
 // Used for RefSeq ID
-export const refseqValidation = new RegExp(
-    /^[a-zA-Z0-9_]+([-.][a-zA-Z0-9_]+)*$/
-  );
+export const refseqValidation = (val) => {
+    return /^[a-zA-Z0-9_]+([-.][a-zA-Z0-9_]+)*$/.test(val)
+  }
 
-// Used for Uniprot ID
-export const uniprotValidation = new RegExp(/^[A-Za-z0-9_]+$/);
+  // Used for Uniprot ID
+export const uniprotValidation = (val) => {
+  return /^[A-Za-z0-9_]+$/.test(val);
+}
 
 // Use for Protein Sequence
-export const proteinValidation = new RegExp(
-    /^[ACDEFGHIKLMNPQRSTVWYacdefghiklmnpqrstvwx]*$/
-    );
-
+export const proteinValidation = (val) => {
+  return /^[ACDEFGHIKLMNPQRSTVWYacdefghiklmnpqrstvwx]*$/.test(val)
```

Comment:
I noticed that this regex isn't producing the intended output, maybe I wrote it wrong?

It should only allow the (ACDEFGHIKLMNPQRSTVWYacdefghiklmnpqrstvwx) characters, in any order.

Reply:
What’s an example you used which broke this validation?

Reply:
Here's an example of the validation working (for the Uniprot id input)
<img width="188" alt="image" src="https://github.com/simonsnitz/snowprint-ui/assets/16387251/408b7415-0532-4ca4-800b-b326755a45d0">